### ISSUE TITLE
MangaParkV3: HTTP error 405

### ISSUE THREAD

**Issue by vncrack** [tachiyomiorg/tachiyomi-extensions#11360]
### Source information

MangaPark v3 1.2.14

### Steps to reproduce

1. Choose MangaPark v3 source
2. Load a manga

### Expected behavior

Chapters load normally

### Actual behavior

Chapters do not load (do not pay attention to the manga, I just randomly chose)

![image](https://user-images.githubusercontent.com/78357927/162249556-83c10391-ee93-4b30-9142-e6f9a8a7adbe.png)

### Tachiyomi version

0.13.1

### Android version

Android 12

### Other details

Latest manga still update but cannot load chapters
Webview works normally

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

**Comment by gan-of-culture** [comment]
I have the same issue.

**Comment by Jaykoerner** [comment]
They have disabled the old v35 version of the site and your getting a redirect to the latest(v36) unfortunately the chapters links are store differently in v36, this was known when I patched it to make it work months ago, unfortunately I am not smart enough to do more then simple edits and this requires a fairly in-depth redesign of the scraper. This will likely be out of commission for a while, 

Anyways regardless of my ineptitude I will attempt to fix it(even if mostly for my own use, if I get something working I will push it though), but this is not a notice to other developers that someone's working on it, since I do not understand Java or kotlin, so any any attempt I make even if working will likely be poorly executed.

**Comment by gan-of-culture** [comment]
Go for it :)

**Comment by Jaykoerner** [comment]
> Go for it :)

Really what needs to be done. But totally outside my abilities enough to even try, is since boto.to and mangapark are both developed by the same company and so similar they should be rolled together as a multi-source, is this feasible?, I don't know, but the differences between the layouts of the two sites is small granted it does exist mostly and how the chapters links need to be scraped, anyways in the process of recording the repo and loading up visual studio to take a look. If I don't post an update in the next couple days I've probably failed....

**Comment by Jaykoerner** [comment]
Discovery update for whoever fixes this the proper way, currently the extension is setup to create and parse a ajax request, the client side library for the site still supports this request, but from the best of my understanding this data is no longer accessible in this format from the server, editing the request to instead fetch the chapter list by latest instead of serial does fetch valid chapters that can be parsed by the current extension, the downside is that the first request only fetches five chapters, you can use a load more request to fetch another set(you will need to do this multiple times sadly) but that  may not be the best option in light of a redesign of the scraper to instead just scrap the full chapter list from the main page, but I do not have enough info about how this deals with multiple languages, this may not be a valid option outside of English, please note this is preliminary information from someone who does not understand enough to know what they don't know it is possible the format of the request has changed and you can still get the required information via the method the extension uses with a differently crafted request...

**Comment by randw0** [comment]
@FourTOne5 any chance you can take a look at this?

**Comment by FourTOne5** [comment]
> @FourTOne5 any chance you can take a look at this?

I had plans to work on this but it'll take some time to complete it

**Comment by Jaykoerner** [comment]
more information for a viable way to scrape the data, using the request ajax.reader.subject.episodes.by.folder, this will require multiple request to be generated, but a known quantity since they can be counted on the main page, it will also require a new variable to be passed along with the request, one that identifies the scanlator/source, this is readily available as a set of numbers next to the source name and the total chapter count.

**Comment by FourTOne5** [comment]
> more information for a viable way to scrape the data, using the request ajax.reader.subject.episodes.by.folder, this will require multiple request to be generated, but a known quantity since they can be counted on the main page, it will also require a new variable to be passed along with the request, one that identifies the scanlator/source, this is readily available as a set of numbers next to the source name and the total chapter count.

Doing by folder will give a ton of duplicate chapters

**Comment by Jaykoerner** [comment]
> > more information for a viable way to scrape the data, using the request ajax.reader.subject.episodes.by.folder, this will require multiple request to be generated, but a known quantity since they can be counted on the main page, it will also require a new variable to be passed along with the request, one that identifies the scanlator/source, this is readily available as a set of numbers next to the source name and the total chapter count.
> 
> Doing by folder will give a ton of duplicate chapters

yeah it will definitely do that, not sure if how its different then now, all sources already got added to the list(well of the users language), you shouldn't have any problem the current way of sorting by source in the app seems to work fine for keeping you from seeing the same chapter more then nessasary.

**Comment by Jaykoerner** [comment]
@FourTOne5 I got a simple scraper running for English only, I should be easily expanded to scrap all languages in theory but without the knowledge of how to filter the output that's the best I can do, every chapter is available on the main document before any extra post requests are made to fetch filtered requests, English is pre filtered into it's own div so it's easy to isolate,


The two issues I have with creating update to the extension that can be published via this method,

1) I do not have the knowledge to turn a plain text string showing the age of the chapter into a Unicode timestamp, that and the data available via pure scraping isn't very specific (examples "1 min ago", "2 hours ago", "3 days ago")

2) I do not know how to filter the multi language chapters in a way useable by tachiyomi,  the chapters can be filtered either by parsing the urls(there is a language designation at the end of the urls) or by using a child objects css class to determine what language it is.


For personal use I have no issues with these limitations but I am not sure how the devs feel

Also if they are fine with English only working and a lack of proper timestamps, till a proper fix can be made by you or another dev, what is the perfered way way to set the time stamp when unknown. I am currently setting it to 0

Sorry for the bother, I only learned some front end html and css and the most limited amounts of JavaScript over 10 years ago and that limits my ability to properly fix issues like this to such a hacked together method.
